Task 2/Task 2.1.2/Task 2.1.2/Program.cs

```diff
@@ -0,0 +1,804 @@
+            {
+                User user = new User(username);
+                users.Add(user);
+                return user;
+            }
+            else 
+            {
+                int index;
+                bool result = SearchUser(username, users, out index);
+                if (result)
+                {
+                    return users[index];
+                }
+                else
+                {
+                    User user = new User(username);
+                    users.Add(user);
+                    return user;
+                }
+            } 
+        }
+
+        /// <summary>
+        /// Метод, который ищет пользователя по имени в списке юзеров. 
+        /// </summary>
+        /// <param name="username"></param>
+        /// <param name="users"></param>
+        /// <param name="index"></param>
+        /// <returns>Если юзер найдет возвращает true и индекс этого пользователя, если нет, то возвращает false  и индекс - 1,
+        /// в случае. если позователь не найдет индекс -1 нигде не применяется и используется в качестве заглушки.</returns>
+        public static bool SearchUser (string username, List <User> users, out int index)
+        {
+            bool flag = false;
+            index = -1;
+            for (int i = 0; i <= users.Count; i++)
+            {
+                if (users[i].Name == username)
+                {
+                    flag = true;
+                    index = i;
+                }
+            }
+
+            return flag;
+        }
+    }
+
+    /// <summary>
+    /// Ниже расположена иерархия наследования фигур. В качестве самого первого и основного параметра, который потом
+    /// наследуют все классы, я взял координаты точки, в которой мы начинаем рисование. В целом, я идею с проверкой 
+    /// по точкам развивать не стал и в большинстве случаев, стал просто оперировать значениями сторон.
+    /// </summary>
+    abstract class Figure
+    {
+        private float _startpointX;
+
+        public float StartpointX
+        {
+            get => _startpointX;
+            set => _startpointX = value;
+        }
+
+        private float _startpointY;
+
+        public float StartpointY
+        {
+            get => _startpointY;
+            set => _startpointY = value;
+        }
+    }
+
+    abstract class OpenFigures : Figure
+    {
+        private float _endlinepointX;
+
+        public float EndlinepointX
+        {
+            get => _endlinepointX;
+            set => _endlinepointX = value;
+        }
+
+        private float _endlinepointY;
+
+        public float EndlinepointY
+        {
+            get => _endlinepointY;
+            set => _endlinepointY = value;
+        }
+        public virtual double Length { get; }
+    }
+
+    abstract class ClosedFigures: Figure
+    {
+        public virtual double GetSquare { get; }
+    }
+
+    abstract class VolumeFigures : ClosedFigures
+    {
+        public virtual double GetVolume { get; }
```

Comment:
`Get` word is for when you do it as a method. In CLR getter for this property will be a `get_GetVolume` method. There's no need to specify that it's a readonly property in such way.